src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -1376,3 +1364,19 @@ def _get_authorization_code(resource, authority_url):
     if results.get('no_browser'):
         raise RuntimeError()
     return results
+
+
+def _transform_subscription_for_multiapi(s, s_dict):
+    """
+    Transforms properties from Subscriptions - List 2019-06-01 and later to the subscription dict.
+
+    :param s: subscription object
+    :param s_dict: subscription dict
+    """
+    if hasattr(s, 'home_tenant_id'):
+        s_dict[_HOME_TENANT_ID] = s.home_tenant_id
+    if hasattr(s, 'managed_by_tenants'):
+        if s.managed_by_tenants is None:
+            s_dict[_MANAGED_BY_TENANTS] = None
```

Comment:
If ARM returns `"managedByTenants": null`, Azure CLI output will also have `"managedByTenants": null`. This behavior doesn't comply with swagger either as `null` is not an array, so users assuming `managedByTenants` to be an array may also face failure.

**Actually, there is no way to develop omnipotent failure recovery for all kinds of REST API mistakes. Making service comply with the swagger is the best and ultimate solution.**

In this case, 

1. `managedByTenants` property MUST exist
2. `managedByTenants` MUST be an array, not `null`

https://learn.microsoft.com/en-us/rest/api/resources/subscriptions/list?tabs=HTTP

![image](https://user-images.githubusercontent.com/4003950/215376977-0f289ea4-f577-4fee-9602-7dd3aeb7ccf9.png)